pkg/api/types/image_types.go

```diff
@@ -165,14 +169,15 @@ type ImagePushOptions struct {
 	IpfsAddress string
 	// Suppress verbose output
 	Quiet bool
+	// If non-nil, the Push job will send upload statuses to the handler instead of Stdout
+	ProgressHandler jobs.StatusHandler
 	// AllowNondistributableArtifacts allow pushing non-distributable artifacts
 	AllowNondistributableArtifacts bool
 }
 
 // ImagePullOptions specifies options for `nerdctl (image) pull`.
 type ImagePullOptions struct {
 	Stdout        io.Writer
```

Comment:
How about just removing `Stdout` and pass `progressHandler` as `jobs.PrintProgress(cmd.OutOrStderr())` in CLI layer?

Reply:
Sounds good to me. The push command also prints the image reference to stdout if `--quiet` flag is set, so I have changed the push/pull method signature to return image reference along with error.